arrows/ceres/register_algorithms.cxx

```diff
@@ -33,26 +33,39 @@
  * \brief Ceres algorithm registration implementation
  */
 
-#include "register_algorithms.h"
+#include <arrows/ceres/kwiver_algo_ceres_export.h>
+#include <vital/algo/algorithm_factory.h>
 
-
-#include <arrows/algorithm_plugin_interface_macros.h>
 #include <arrows/ceres/bundle_adjust.h>
 
 
 namespace kwiver {
 namespace arrows {
 namespace ceres {
 
-/// Register Ceres algorithm implementations with the given or global registrar
-int register_algorithms( vital::registrar &reg )
+extern "C"
+KWIVER_ALGO_CERES_EXPORT
+void
+register_factories( kwiver::vital::plugin_loader& vpm )
 {
-  REGISTRATION_INIT( reg );
+  static auto const module_name = std::string( "algorithms.ceres" );
+  if (vpm.is_module_loaded( module_name ) )
+  {
+    return;
+  }
+
+  // add factory               implementation-name       type-to-create
+  auto fact = vpm.ADD_ALGORITHM( "ceres", kwiver::arrows::ceres::bundle_adjust );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "Attempts short-term loop closure based on percentage of feature "
```

Comment:
Looks like a description copied from another algorithm.  It should be something like this:

    Uses Ceres Solver to bundle adjust camera and landmark parameters